src/wizard/hints.jl

```diff
@@ -41,8 +41,16 @@ function provide_hints(state::WizardState, path::AbstractString)
     for (root, dirs, files) in walkdir(path)
         for file in files
             file_path = joinpath(root, file)
-            contents = String(read(file_path))
-            if file == "configure" && contains(contents, "Generated by GNU Autoconf")
+
+            # Helper function to try to read the given path's contents, but
+            # returning an empty string on error (for e.g. broken symlinks)
+            read_contents = (path) -> try
```

Comment:
Why not `read_contents(path) = try ...`?

Reply:
You could also move it out of the loop, since it doesn't need to be regenerated with every iteration

Reply:
Is there a reason to prefer `foo(x) = ...` instead of `foo = x -> ...`?

Reply:
The reason I have it within the loop is because I like to have it right next to the only place it's being used; reduce mental strain on whoever is trying to read this code.  To really squeeze that last 0.1% performance out of this loop, let's just hope that the compiler realizes this value is a constant and hoists it out of the loop.  ;)

Reply:
Stylistically it's odd to use an anonymous function if you're going to name it.